backend/src/utils/prometheusUtils.ts

```diff
@@ -1,30 +1,40 @@
-import { KubeFastifyInstance, OauthFastifyRequest, PrometheusResponse } from '../types';
+import {
+  KubeFastifyInstance,
+  OauthFastifyRequest,
+  PrometheusQueryRangeResponse,
+  PrometheusQueryResponse,
+} from '../types';
 import https from 'https';
 import { getDirectCallOptions } from './directCallUtils';
 import { DEV_MODE } from './constants';
 
-export const callPrometheus = async (
+const callPrometheus = async <T>(
   fastify: KubeFastifyInstance,
   request: OauthFastifyRequest,
   query: string,
-  namespace: string,
-): Promise<{ code: number; response: PrometheusResponse }> => {
+  host: string,
+  queryRange?: boolean,
+): Promise<{ code: number; response: T }> => {
   if (!query) {
     fastify.log.warn('Prometheus call was made without a query');
     return Promise.reject({ code: 400, response: 'Failed to provide a query' });
   }
 
-  // Use a local path to the thanos querier; only works on-cluster
-  let host = `https://thanos-querier.openshift-monitoring.svc.cluster.local:9091`;
-  if (DEV_MODE) {
-    const apiPath = fastify.kube.config.getCurrentCluster().server;
-    const namedHost = apiPath.slice('https://api.'.length).split(':')[0];
-    host = `https://thanos-querier-openshift-monitoring.apps.${namedHost}`;
+  if (!host) {
+    fastify.log.warn('Prometheus call was made with a host that does not exist');
+    return Promise.reject({ code: 400, response: 'Failed to find the prometheus instance host' });
   }
 
-  const url = `${host}/api/v1/query?namespace=${namespace}&query=${query}`;
+  const url = `${host}/api/v1/${queryRange ? 'query_range' : 'query'}?${query}`;
```

Comment:
Small nit -- instead of making this a boolean -- do you think we could use an enum? Makes the call signature look better -- instead of `true` it's `QueryType.QUERY_RANGE`. And you can default it so `QueryType.QUERY` if you want to keep the same call signature for the other prometheus call.